tools/chplcheck/src/rule_types.py

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright 2024 Hewlett Packard Enterprise Development LP
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import typing
+from dataclasses import dataclass
+
+import chapel
+from fixits import ChapelFixit
+
+
+@dataclass
+class FullBasicRuleResult:
+    check: bool
```

Comment:
Noted below, I think `check` here can be removed since, why would you recommend a fixit if nothing's wrong?